### PR TITLE
VPC/Subnet module

### PR DESCRIPTION
**Pull request by phoolish**

### CODE REVIEW

File: vpc/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output "vpc_id" {
+  value = "${aws_vpc.mod.id}"
+}
+
+output "internet_gateway_id" {
+  value = "${aws_internet_gateway.mod.id}"
+}
```
Comment (pturley) [inline vpc/outputs.tf:7]: probably should add cidr as an output

File: vpc/subnets/variables.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+variable "private_netnum_offset" {
+  default = 10
+}
+variable "public_newbits" {
+  default = 8
+}
+variable "public_netnum_offset" {
+  default = 0
+}
+variable "az_to_netnum" {
```
Comment (pturley) [inline vpc/subnets/variables.tf]: I dont see any reason to allow someone to overwrite this

File: vpc/subnets/variables.tf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+variable "private_newbits" {
+  default = 8
+}
+variable "private_netnum_offset" {
+  default = 10
+}
+variable "public_newbits" {
+  default = 8
+}
+variable "public_netnum_offset" {
```
Comment (pturley) [inline vpc/subnets/variables.tf]: these four variable are confusing as hell. Anyway we could abstract this to make it a bit more user friendly?

File: vpc/subnets/variables.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
+variable "environment_name" { }
```
Comment (pturley) [inline vpc/subnets/variables.tf]: maybe this should just be name

File: vpc/subnets/main.tf [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# Send all traffic over the nat gateway
+resource "aws_route_table" "mod_private" {
+  vpc_id = "${var.vpc_id}"
+  tags { Name = "${var.environment_name}-private" }
+}
+
+# Sets up the private subnets
+resource "aws_subnet" "mod_private" {
+  vpc_id = "${var.vpc_id}"
+  cidr_block = "${cidrsubnet(var.vpc_cidr_block, var.private_newbits, var.private_netnum_offset + lookup(var.az_to_netnum, substr(element(data.aws_availability_zones.available.names, count.index), -1, 1)))}"
```
Comment (stephen-puiszis) [inline vpc/subnets/main.tf]: can we do some of this variable assignment on multiple lines? Might make it more readable/understandable.

File: vpc/subnets/variables.tf (hunk 2 above)
Comment (phoolish) [reply vpc/subnets/variables.tf]: Is there .a way to lock variables?

File: vpc/subnets/variables.tf (hunk 3 above)
Comment (phoolish) [reply vpc/subnets/variables.tf]: Not sure. Suggestions? We could turn it into a single hash for public and private. Plus, I think we should add descriptions to to all variables as a best practice.

File: vpc/subnets/main.tf (hunk 5 above)
Comment (phoolish) [reply vpc/subnets/main.tf]: My thoughts exactly! I was thinking about moving this to a template_file.

File: vpc/outputs.tf (hunk 1 above)
Comment (phoolish) [reply vpc/outputs.tf:7]: agreed.

File: vpc/subnets/variables.tf (hunk 2 above)
Comment (phoolish) [reply vpc/subnets/variables.tf]: wooo https://www.terraform.io/docs/configuration/locals.html

Comment (pturley) [reply vpc/subnets/variables.tf]: Does this help? https://www.terraform.io/docs/configuration/locals.html

File: vpc/subnets/variables.tf (hunk 3 above)
Comment (pturley) [reply vpc/subnets/variables.tf]: I like them as separate variables, but newbits and netnum dont really mean anything to me. No suggestion today. I might have an epiphany later though

File: vpc/subnets/main.tf (hunk 5 above)
Comment (pturley) [reply vpc/subnets/main.tf]: https://www.terraform.io/docs/configuration/locals.html might help here too

Comment (phoolish) [reply vpc/subnets/main.tf]: I tried locals initially, but needing to iterate over the entire map makes it tough.

**Review by pturley — APPROVED** [review]

**Pull request merged**